src/colvarproxy_io.cpp

```diff
@@ -228,6 +229,54 @@ std::istream &colvarproxy_io::input_stream(std::string const &input_name,
 }
 
 
+std::istream &
+colvarproxy_io::input_stream_from_string(std::string const &input_name,
+                                         std::string const &content,
+                                         std::string const description)
+{
+  if (!io_available()) {
+    cvm::error("Error: trying to access an input file/channel "
+               "from the wrong thread.\n", COLVARS_BUG_ERROR);
+    return *input_stream_error_;
+  }
+
+  if (colvarproxy_io::input_stream_exists(input_name)) {
+
+    std::istringstream *iss =
+      dynamic_cast<std::istringstream *>(input_streams_[input_name]);
+    if (iss) {
+      // If there is already a stringstream, replace it
+      delete iss;
+    } else {
+      std::ifstream *ifs =
+        dynamic_cast<std::ifstream *>(input_streams_[input_name]);
+      if (ifs) {
+        if (ifs->is_open()) {
+          ifs->close();
+        }
+      }
+    }
+  }
+
+  input_streams_[input_name] = new std::istringstream(content);
+
+  return *(input_streams_[input_name]);
+}
+
+
+std::istream &
+colvarproxy_io::input_string_stream(std::string const &input_name)
```

Comment:
That's true, but the initialization can be ambiguous because of the optional arguments.

If you don't see a problem, I'm happy to make it less pedantic.